packages/web-app/src/hooks/useCachedDao.tsx

```diff
@@ -0,0 +1,102 @@
+import {isAddress} from 'ethers/lib/utils';
+import {useCache} from './useCache';
+import {useClient} from './useClient';
+import {
+  ICreateDaoERC20Voting,
+  ICreateDaoWhitelistVoting,
+} from '@aragon/sdk-client';
+import {Dao} from 'utils/types';
+
+interface useDaoResponse {
+  update: (dao: Dao) => Dao;
+  createErc20: (dao: ICreateDaoERC20Voting) => Promise<string>;
+  createWhitelist: (dao: ICreateDaoWhitelistVoting) => Promise<string>;
+  get: (address: string) => Dao;
+}
+
+export const useDao = (): useDaoResponse => {
+  const {get: getCache, set: setCache} = useCache();
+  const {erc20, whitelist} = useClient();
+  const get = (address: string): Dao => {
+    if (!isAddress(address)) {
+      throw Error('invalid address when trying to get dao');
+    }
+    const cacheKey = `dao-${address}`;
+    const cacheValue = getCache(cacheKey);
+    // If dao is in cache return it
+    if (cacheValue) {
+      return cacheValue;
+    }
+    // client.getDao(address).then((value) => setCache(cacheKey, value)).catch((e)=> throw Error(e))
+    const value: Dao = {
+      address,
+    };
+    setCache(cacheKey, value);
+    return value;
+  };
+  const createErc20 = (dao: ICreateDaoERC20Voting): Promise<string> => {
+    return new Promise((resolve, reject): void => {
+      if (erc20) {
+        erc20.dao
+          .create(dao)
+          .then((address: string) => {
+            // if creation went correctly should return the dao address
+            // this will be used to identify the dao in the cache
+            const cacheKey = `dao-${address}`;
+            const cacheDao: Dao = {
+              address,
+            };
+            setCache(cacheKey, cacheDao);
+            resolve(address);
+          })
+          .catch((e: Error) => {
+            console.log(e);
+            // thow error if it fails
+            reject(e);
+          });
+      } else {
+        reject('ERC20 SDK client is not initialized correctly');
```

Comment:
Promise rejections should return an `Error` object, not just a string
```suggestion
        reject(new Error('ERC20 SDK client is not initialized correctly'));
```